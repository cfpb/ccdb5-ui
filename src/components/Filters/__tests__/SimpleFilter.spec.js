import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { IntlProvider } from 'react-intl';
import ReduxSimpleFilter, { mapStateToProps } from '../SimpleFilter';

import renderer from 'react-test-renderer';

/**
 *
 * @param initialAggs
 */
function setupSnapshot(initialAggs) {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({
    query: {},
    aggs: {
      company_response: initialAggs,
      timely: 'yes',
    },
  });

  return renderer.create(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ReduxSimpleFilter fieldName="company_response" title="nana" />
      </IntlProvider>
    </Provider>
  );
}

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot([{ key: 'foo', doc_count: 99 }]);
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  let state, ownProps;
  beforeEach(() => {
    state = {
      aggs: {
        foo: [1, 2, 3, 4, 5, 6],
      },
      query: {
        foo: [1],
      },
    };
    ownProps = {
      fieldName: 'foo',
    };
  });

  it('shows if there are any active children', () => {
    const actual = mapStateToProps(state, ownProps);
    expect(actual).toEqual({
      options: [1, 2, 3, 4, 5, 6],
      hasChildren: true,
    });
  });

  it('hides if there are no active children', () => {
    state.query.foo = [];

    const actual = mapStateToProps(state, ownProps);
    expect(actual).toEqual({
      options: [1, 2, 3, 4, 5, 6],
      hasChildren: false,
    });
  });
});
