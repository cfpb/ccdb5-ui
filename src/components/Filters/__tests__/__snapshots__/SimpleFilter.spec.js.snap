// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state renders without crashing 1`] = `
<section
  className="o-expandable aggregation simple company_response"
>
  <button
    aria-label="Hide nana filter"
    className="o-expandable_header o-expandable_target o-expandable_target__collapsed"
    onClick={[Function]}
  >
    <h3
      className="h4 o-expandable_header-left o-expandable_label"
    >
      nana
    </h3>
    <span
      className="o-expandable_header-right o-expandable_link"
    >
      <span
        className="o-expandable_cue o-expandable_cue-open"
      >
        <span
          className="u-visually-hidden-on-mobile"
        >
          Show
        </span>
        <svg
          className="cf-icon-svg"
          viewBox="0 0 1000 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm263.1 550.7H549.6v213.6c0 27.6-22.4 50-50 50s-50-22.4-50-50V655.9H236c-27.6 0-50-22.4-50-50s22.4-50 50-50h213.6V342.3c0-27.6 22.4-50 50-50s50 22.4 50 50v213.6h213.6c27.6 0 50 22.4 50 50s-22.5 50-50.1 50z"
          />
        </svg>
      </span>
    </span>
  </button>
</section>
`;
